resource-managers/kubernetes/core/src/main/scala/org/apache/spark/deploy/k8s/Config.scala

```diff
@@ -417,6 +417,17 @@ private[spark] object Config extends Logging {
       .booleanConf
       .createWithDefault(false)
 
+  val KUBERNETES_EXECUTOR_MISSING_POD_DETECT_DELTA =
+    ConfigBuilder("spark.kubernetes.executor.missingPodDetectDelta")
+      .doc("When a registered executor's POD is missing from the Kubernetes API server's polled " +
+        "list of PODs then this delta time is taken as the accepted time difference between the " +
+        "registration time and the time of the polling. After this time the POD is considered " +
+        "missing from the cluster and the executor will be removed.")
+      .version("3.1.1")
+      .timeConf(TimeUnit.MILLISECONDS)
+      .checkValue(delay => delay > 0, "delay must be a positive time value")
```

Comment:
nit: remove time as it simply has to be a positive value (millis assumed, no?)

Reply:
I do not think so.  Negative value is parsed and accepted as valid value, call stack:

https://github.com/apache/spark/blob/3afa99ae6393e9a3be88c24aa764e793839eec02/core/src/main/scala/org/apache/spark/internal/config/ConfigBuilder.scala#L252-L255

https://github.com/apache/spark/blob/3afa99ae6393e9a3be88c24aa764e793839eec02/core/src/main/scala/org/apache/spark/internal/config/ConfigBuilder.scala#L56

**See the optional "-":**

https://github.com/apache/spark/blob/3afa99ae6393e9a3be88c24aa764e793839eec02/common/network-common/src/main/java/org/apache/spark/network/util/JavaUtils.java#L232

https://github.com/apache/spark/blob/3afa99ae6393e9a3be88c24aa764e793839eec02/common/network-common/src/main/java/org/apache/spark/network/util/JavaUtils.java#L237

Reply:
Again, we should not make such a difference which exists in only branch-3.0, @jaceklaskowski .